`efi_disk` defaults to invalid `file_format`

**Describe the bug**
When making a vm with an `efi_disk`, the `file_format` default to `qcow` which causes errors. I think the only valid value for an `efi_disk` is `raw`.

**To Reproduce**
Steps to reproduce the behavior:

1. Create an `efi_disk` without specifying `file_format` (defaults to `qcow`)
2. Run `terraform apply`
3. See error
4. Specify `file_format = "raw"`
5. Run `terraform apply`
6. No error this time

Please also provide a minimal Terraform configuration that reproduces the issue.

```hcl

resource "proxmox_virtual_environment_vm" "example" {
  //...

  efi_disk {
    datastore_id = "datastore-id"
    type = "4m"
    // file_format = "raw"
  }

  //...
}
```

**Expected behavior**
`efi_disk` should default to `raw`

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Add any other context about the problem here.

- Single or clustered Proxmox: Single
- Provider version (ideally it should be the latest version): v0.54.0
- Terraform version: OpenTofu v1.6.0
- OS (where you run Terraform from): NixOS
- Debug logs (`TF_LOG=DEBUG terraform apply`):

<details>
<summary>Debug logs</summary>

```
OpenTofu used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

OpenTofu will perform the following actions:

  # proxmox_virtual_environment_vm.nixos_vm will be created
  + resource "proxmox_virtual_environment_vm" "nixos_vm" {
      + acpi                    = true
      + bios                    = "ovmf"
      + id                      = (known after apply)
      + ipv4_addresses          = (known after apply)
      + ipv6_addresses          = (known after apply)
      + keyboard_layout         = "en-us"
      + mac_addresses           = (known after apply)
      + machine                 = "q35"
      + migrate                 = false
      + name                    = "nixos.local"
      + network_interface_names = (known after apply)
      + node_name               = "elodin"
      + on_boot                 = true
      + protection              = false
      + reboot                  = false
      + scsi_hardware           = "virtio-scsi-pci"
      + started                 = true
      + stop_on_destroy         = false
      + tablet_device           = true
      + template                = false
      + timeout_clone           = 1800
      + timeout_create          = 1800
      + timeout_migrate         = 1800
      + timeout_move_disk       = 1800
      + timeout_reboot          = 1800
      + timeout_shutdown_vm     = 1800
      + timeout_start_vm        = 1800
      + timeout_stop_vm         = 300
      + vm_id                   = 100

      + agent {
          + enabled = true
          + timeout = "15m"
          + trim    = false
          + type    = "virtio"
        }

      + cpu {
          + architecture = "x86_64"
          + cores        = 4
          + hotplugged   = 0
          + limit        = 0
          + numa         = false
          + sockets      = 1
          + type         = "host"
          + units        = 1024
        }

      + disk {
          + aio               = "io_uring"
          + backup            = true
          + cache             = "none"
          + datastore_id      = "local-zfs"
          + discard           = "ignore"
          + file_format       = (known after apply)
          + file_id           = "local:iso/nixos_base_2.img"
          + interface         = "scsi0"
          + iothread          = false
          + path_in_datastore = (known after apply)
          + replicate         = true
          + size              = 208
          + ssd               = false
        }

      + efi_disk {
          + datastore_id      = "local-zfs"
          + file_format       = (known after apply)
          + pre_enrolled_keys = false
          + type              = "4m"
        }

      + memory {
          + dedicated      = 4096
          + floating       = 0
          + keep_hugepages = false
          + shared         = 0
        }

      + network_device {
          + bridge      = "vmbr0"
          + enabled     = true
          + firewall    = false
          + mac_address = (known after apply)
          + model       = "virtio"
          + mtu         = 0
          + queues      = 0
          + rate_limit  = 0
          + vlan_id     = 0
        }

      + operating_system {
          + type = "l26"
        }

      + serial_device {
          + device = "socket"
        }

      + tpm_state {
          + datastore_id = "local-zfs"
          + version      = "v2.0"
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

Saved the plan to: tf_plan

To perform exactly these actions, run the following command to apply:
    tofu apply "tf_plan"
2024-04-28T11:55:58.817-0700 [INFO]  OpenTofu version: 1.6.0
2024-04-28T11:55:58.817-0700 [DEBUG] using github.com/hashicorp/go-tfe v1.36.0
2024-04-28T11:55:58.817-0700 [DEBUG] using github.com/hashicorp/hcl/v2 v2.19.1
2024-04-28T11:55:58.817-0700 [DEBUG] using github.com/hashicorp/terraform-svchost v0.1.1
2024-04-28T11:55:58.817-0700 [DEBUG] using github.com/zclconf/go-cty v1.14.1
2024-04-28T11:55:58.817-0700 [INFO]  Go runtime version: go1.21.5
2024-04-28T11:55:58.817-0700 [INFO]  CLI args: []string{"/nix/store/ljypgfyyqfb63826l65cpwky90hx139x-opentofu-1.6.0/bin/tofu", "apply", "tf_plan"}
2024-04-28T11:55:58.817-0700 [DEBUG] Attempting to open CLI config file: /home/andrewhamon/.tofurc
2024-04-28T11:55:58.817-0700 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2024-04-28T11:55:58.817-0700 [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
2024-04-28T11:55:58.817-0700 [DEBUG] ignoring non-existing provider search directory /home/andrewhamon/.terraform.d/plugins
2024-04-28T11:55:58.817-0700 [DEBUG] ignoring non-existing provider search directory /home/andrewhamon/.local/share/terraform/plugins
2024-04-28T11:55:58.817-0700 [DEBUG] ignoring non-existing provider search directory /nix/store/h3ayxf5dk1gdf3s5716fh1ys815qxxag-patchelf-0.15.0/share/terraform/plugins
2024-04-28T11:55:58.817-0700 [DEBUG] ignoring non-existing provider search directory /nix/store/wnblprh9a171idncjqcba2l5hdv0bl79-desktops/share/terraform/plugins
2024-04-28T11:55:58.817-0700 [DEBUG] ignoring non-existing provider search directory /home/andrewhamon/.nix-profile/share/terraform/plugins
2024-04-28T11:55:58.817-0700 [DEBUG] ignoring non-existing provider search directory /nix/profile/share/terraform/plugins
2024-04-28T11:55:58.817-0700 [DEBUG] ignoring non-existing provider search directory /home/andrewhamon/.local/state/nix/profile/share/terraform/plugins
2024-04-28T11:55:58.817-0700 [DEBUG] ignoring non-existing provider search directory /etc/profiles/per-user/andrewhamon/share/terraform/plugins
2024-04-28T11:55:58.817-0700 [DEBUG] ignoring non-existing provider search directory /nix/var/nix/profiles/default/share/terraform/plugins
2024-04-28T11:55:58.817-0700 [DEBUG] ignoring non-existing provider search directory /run/current-system/sw/share/terraform/plugins
2024-04-28T11:55:58.817-0700 [DEBUG] Found the config directory: /home/andrewhamon/.terraform.d
2024-04-28T11:55:58.817-0700 [INFO]  CLI command args: []string{"apply", "tf_plan"}
2024-04-28T11:55:58.830-0700 [DEBUG] checking for provisioner in "."
2024-04-28T11:55:58.830-0700 [DEBUG] checking for provisioner in "/nix/store/ljypgfyyqfb63826l65cpwky90hx139x-opentofu-1.6.0/bin"
2024-04-28T11:55:58.831-0700 [INFO]  backend/local: starting Apply operation
2024-04-28T11:55:58.833-0700 [DEBUG] created provider logger: level=debug
2024-04-28T11:55:58.833-0700 [INFO]  provider: configuring client automatic mTLS
2024-04-28T11:55:58.840-0700 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.opentofu.org/bpg/proxmox/0.54.0/linux_amd64/terraform-provider-proxmox_v0.54.0 args=[".terraform/providers/registry.opentofu.org/bpg/proxmox/0.54.0/linux_amd64/terraform-provider-proxmox_v0.54.0"]
2024-04-28T11:55:58.841-0700 [DEBUG] provider: plugin started: path=.terraform/providers/registry.opentofu.org/bpg/proxmox/0.54.0/linux_amd64/terraform-provider-proxmox_v0.54.0 pid=3851550
2024-04-28T11:55:58.841-0700 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.opentofu.org/bpg/proxmox/0.54.0/linux_amd64/terraform-provider-proxmox_v0.54.0
2024-04-28T11:55:58.844-0700 [INFO]  provider.terraform-provider-proxmox_v0.54.0: configuring server automatic mTLS: timestamp=2024-04-28T11:55:58.843-0700
2024-04-28T11:55:58.854-0700 [DEBUG] provider.terraform-provider-proxmox_v0.54.0: plugin address: address=/tmp/nix-shell.F4wUPY/plugin1763277481 network=unix timestamp=2024-04-28T11:55:58.854-0700
2024-04-28T11:55:58.854-0700 [DEBUG] provider: using plugin: version=6
2024-04-28T11:55:58.867-0700 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2024-04-28T11:55:58.868-0700 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.opentofu.org/bpg/proxmox/0.54.0/linux_amd64/terraform-provider-proxmox_v0.54.0 pid=3851550
2024-04-28T11:55:58.868-0700 [DEBUG] provider: plugin exited
2024-04-28T11:55:58.868-0700 [INFO]  backend/local: apply calling Apply
2024-04-28T11:55:58.868-0700 [DEBUG] Building and walking apply graph for NormalMode plan
2024-04-28T11:55:58.869-0700 [DEBUG] Resource state not found for node "proxmox_virtual_environment_vm.nixos_vm", instance proxmox_virtual_environment_vm.nixos_vm
2024-04-28T11:55:58.869-0700 [DEBUG] ProviderTransformer: "proxmox_virtual_environment_vm.nixos_vm (expand)" (*tofu.nodeExpandApplyableResource) needs provider["registry.opentofu.org/bpg/proxmox"]
2024-04-28T11:55:58.869-0700 [DEBUG] ProviderTransformer: "proxmox_virtual_environment_vm.nixos_vm_2 (expand)" (*tofu.nodeExpandApplyableResource) needs provider["registry.opentofu.org/bpg/proxmox"]
2024-04-28T11:55:58.869-0700 [DEBUG] ProviderTransformer: "proxmox_virtual_environment_vm.nixos_vm" (*tofu.NodeApplyableResourceInstance) needs provider["registry.opentofu.org/bpg/proxmox"]
2024-04-28T11:55:58.869-0700 [DEBUG] ProviderTransformer: "proxmox_virtual_environment_file.nixos_base (expand)" (*tofu.nodeExpandApplyableResource) needs provider["registry.opentofu.org/bpg/proxmox"]
2024-04-28T11:55:58.869-0700 [DEBUG] ProviderTransformer: "proxmox_virtual_environment_file.nixos_base_lxc (expand)" (*tofu.nodeExpandApplyableResource) needs provider["registry.opentofu.org/bpg/proxmox"]
2024-04-28T11:55:58.869-0700 [DEBUG] ReferenceTransformer: "proxmox_virtual_environment_file.nixos_base (expand)" references: []
2024-04-28T11:55:58.869-0700 [DEBUG] ReferenceTransformer: "proxmox_virtual_environment_file.nixos_base_lxc (expand)" references: []
2024-04-28T11:55:58.869-0700 [DEBUG] ReferenceTransformer: "proxmox_virtual_environment_vm.nixos_vm (expand)" references: [proxmox_virtual_environment_file.nixos_base (expand)]
2024-04-28T11:55:58.869-0700 [DEBUG] ReferenceTransformer: "proxmox_virtual_environment_vm.nixos_vm_2 (expand)" references: [proxmox_virtual_environment_file.nixos_base (expand)]
2024-04-28T11:55:58.869-0700 [INFO]  ReferenceTransformer: reference not found: "proxmox_virtual_environment_file.nixos_base#destroy"
2024-04-28T11:55:58.869-0700 [DEBUG] ReferenceTransformer: "proxmox_virtual_environment_vm.nixos_vm" references: [proxmox_virtual_environment_file.nixos_base (expand)]
2024-04-28T11:55:58.869-0700 [DEBUG] ReferenceTransformer: "provider[\"registry.opentofu.org/bpg/proxmox\"]" references: []
2024-04-28T11:55:58.869-0700 [DEBUG] pruneUnusedNodes: proxmox_virtual_environment_file.nixos_base_lxc (expand) is no longer needed, removing
2024-04-28T11:55:58.869-0700 [DEBUG] pruneUnusedNodes: proxmox_virtual_environment_vm.nixos_vm_2 (expand) is no longer needed, removing
2024-04-28T11:55:58.869-0700 [DEBUG] Starting graph walk: walkApply
2024-04-28T11:55:58.870-0700 [DEBUG] created provider logger: level=debug
2024-04-28T11:55:58.870-0700 [INFO]  provider: configuring client automatic mTLS
2024-04-28T11:55:58.873-0700 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.opentofu.org/bpg/proxmox/0.54.0/linux_amd64/terraform-provider-proxmox_v0.54.0 args=[".terraform/providers/registry.opentofu.org/bpg/proxmox/0.54.0/linux_amd64/terraform-provider-proxmox_v0.54.0"]
2024-04-28T11:55:58.874-0700 [DEBUG] provider: plugin started: path=.terraform/providers/registry.opentofu.org/bpg/proxmox/0.54.0/linux_amd64/terraform-provider-proxmox_v0.54.0 pid=3851566
2024-04-28T11:55:58.874-0700 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.opentofu.org/bpg/proxmox/0.54.0/linux_amd64/terraform-provider-proxmox_v0.54.0
2024-04-28T11:55:58.876-0700 [INFO]  provider.terraform-provider-proxmox_v0.54.0: configuring server automatic mTLS: timestamp=2024-04-28T11:55:58.876-0700
2024-04-28T11:55:58.887-0700 [DEBUG] provider.terraform-provider-proxmox_v0.54.0: plugin address: address=/tmp/nix-shell.F4wUPY/plugin1647313817 network=unix timestamp=2024-04-28T11:55:58.887-0700
2024-04-28T11:55:58.887-0700 [DEBUG] provider: using plugin: version=6
2024-04-28T11:55:58.897-0700 [INFO]  provider.terraform-provider-proxmox_v0.54.0: Configuring the Proxmox provider...: tf_req_id=c46b33d0-0499-b4f2-aaf5-5a4908e3784c tf_provider_addr=registry.terraform.io/bpg/proxmox @caller=github.com/bpg/terraform-provider-proxmox/fwprovider/provider.go:233 @module=proxmox tf_mux_provider="*proto6server.Server" tf_rpc=ConfigureProvider timestamp=2024-04-28T11:55:58.897-0700
2024-04-28T11:55:58.907-0700 [WARN]  Provider "registry.opentofu.org/bpg/proxmox" produced an invalid plan for proxmox_virtual_environment_vm.nixos_vm, but we are tolerating it because it is using the legacy plugin SDK.
    The following problems may be the cause of any confusing errors from downstream operations:
      - .acpi: planned value cty.True for a non-computed attribute
      - .on_boot: planned value cty.True for a non-computed attribute
      - .scsi_hardware: planned value cty.StringVal("virtio-scsi-pci") for a non-computed attribute
      - .stop_on_destroy: planned value cty.False for a non-computed attribute
      - .migrate: planned value cty.False for a non-computed attribute
      - .timeout_reboot: planned value cty.NumberIntVal(1800) for a non-computed attribute
      - .template: planned value cty.False for a non-computed attribute
      - .timeout_clone: planned value cty.NumberIntVal(1800) for a non-computed attribute
      - .protection: planned value cty.False for a non-computed attribute
      - .timeout_create: planned value cty.NumberIntVal(1800) for a non-computed attribute
      - .timeout_stop_vm: planned value cty.NumberIntVal(300) for a non-computed attribute
      - .tablet_device: planned value cty.True for a non-computed attribute
      - .timeout_shutdown_vm: planned value cty.NumberIntVal(1800) for a non-computed attribute
      - .keyboard_layout: planned value cty.StringVal("en-us") for a non-computed attribute
      - .started: planned value cty.True for a non-computed attribute
      - .timeout_move_disk: planned value cty.NumberIntVal(1800) for a non-computed attribute
      - .reboot: planned value cty.False for a non-computed attribute
      - .timeout_migrate: planned value cty.NumberIntVal(1800) for a non-computed attribute
      - .timeout_start_vm: planned value cty.NumberIntVal(1800) for a non-computed attribute
      - .efi_disk[0].pre_enrolled_keys: planned value cty.False for a non-computed attribute
      - .memory[0].floating: planned value cty.NumberIntVal(0) for a non-computed attribute
      - .memory[0].keep_hugepages: planned value cty.False for a non-computed attribute
      - .memory[0].shared: planned value cty.NumberIntVal(0) for a non-computed attribute
      - .cpu[0].limit: planned value cty.NumberIntVal(0) for a non-computed attribute
      - .cpu[0].numa: planned value cty.False for a non-computed attribute
      - .cpu[0].sockets: planned value cty.NumberIntVal(1) for a non-computed attribute
      - .cpu[0].units: planned value cty.NumberIntVal(1024) for a non-computed attribute
      - .cpu[0].architecture: planned value cty.StringVal("x86_64") for a non-computed attribute
      - .cpu[0].hotplugged: planned value cty.NumberIntVal(0) for a non-computed attribute
      - .disk[0].cache: planned value cty.StringVal("none") for a non-computed attribute
      - .disk[0].replicate: planned value cty.True for a non-computed attribute
      - .disk[0].ssd: planned value cty.False for a non-computed attribute
      - .disk[0].aio: planned value cty.StringVal("io_uring") for a non-computed attribute
      - .disk[0].discard: planned value cty.StringVal("ignore") for a non-computed attribute
      - .disk[0].iothread: planned value cty.False for a non-computed attribute
      - .disk[0].backup: planned value cty.True for a non-computed attribute
      - .agent[0].timeout: planned value cty.StringVal("15m") for a non-computed attribute
      - .agent[0].trim: planned value cty.False for a non-computed attribute
      - .agent[0].type: planned value cty.StringVal("virtio") for a non-computed attribute
      - .serial_device[0].device: planned value cty.StringVal("socket") for a non-computed attribute
      - .network_device[0].enabled: planned value cty.True for a non-computed attribute
      - .network_device[0].queues: planned value cty.NumberIntVal(0) for a non-computed attribute
      - .network_device[0].rate_limit: planned value cty.NumberIntVal(0) for a non-computed attribute
      - .network_device[0].vlan_id: planned value cty.NumberIntVal(0) for a non-computed attribute
      - .network_device[0].firewall: planned value cty.False for a non-computed attribute
      - .network_device[0].model: planned value cty.StringVal("virtio") for a non-computed attribute
      - .network_device[0].mtu: planned value cty.NumberIntVal(0) for a non-computed attribute
proxmox_virtual_environment_vm.nixos_vm: Creating...
2024-04-28T11:55:58.908-0700 [INFO]  Starting apply for proxmox_virtual_environment_vm.nixos_vm
2024-04-28T11:55:58.908-0700 [DEBUG] proxmox_virtual_environment_vm.nixos_vm: applying the planned Create change
2024-04-28T11:55:58.909-0700 [DEBUG] provider.terraform-provider-proxmox_v0.54.0: [DEBUG] setting computed for "ipv6_addresses" from ComputedKeys
2024-04-28T11:55:58.910-0700 [DEBUG] provider.terraform-provider-proxmox_v0.54.0: [DEBUG] setting computed for "ipv4_addresses" from ComputedKeys
2024-04-28T11:55:58.910-0700 [DEBUG] provider.terraform-provider-proxmox_v0.54.0: [DEBUG] setting computed for "mac_addresses" from ComputedKeys
2024-04-28T11:55:58.910-0700 [DEBUG] provider.terraform-provider-proxmox_v0.54.0: [DEBUG] setting computed for "network_interface_names" from ComputedKeys
2024-04-28T11:55:58.911-0700 [DEBUG] provider.terraform-provider-proxmox_v0.54.0: Sending HTTP Request: tf_http_req_method=POST tf_http_req_version=HTTP/1.1 tf_http_trans_id=acaf6166-bf97-17be-8871-3e7d8e5b6c4a tf_mux_provider=tf5to6server.v5tov6Server Authorization="PVEAPIToken=root@pam!opentofu=da48a39e-2015-4af6-a3d0-94ae09874e74" Accept-Encoding=gzip tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/logging/logging_http_transport.go:162 Accept=application/json User-Agent=Go-http-client/1.1 tf_http_req_uri=/api2/json/nodes/elodin/qemu tf_req_id=9f849788-b263-09c2-db59-09a8b616d506 tf_resource_type=proxmox_virtual_environment_vm @module=proxmox Content-Length=541 Content-Type=application/x-www-form-urlencoded Host=10.69.13.2:8006 tf_http_op_type=request tf_http_req_body="acpi=1&agent=enabled%3D1%2Cfstrim_cloned_disks%3D0%2Ctype%3Dvirtio&balloon=0&bios=ovmf&boot=order%3Dscsi0%3Bnet0&cores=4&cpu=cputype%3Dhost&cpuunits=1024&efidisk0=file%3Dlocal-zfs%3A1%2Cformat%3Dqcow2%2Cefitype%3D4m%2Cpre-enrolled-keys%3D0&keyboard=en-us&machine=q35&memory=4096&name=nixos.local&net0=model%3Dvirtio%2Cbridge%3Dvmbr0%2Cfirewall%3D0&numa=0&onboot=1&ostype=l26&protection=0&scsihw=virtio-scsi-pci&serial0=socket&sockets=1&tablet=1&template=0&tpmstate0=file%3Dlocal-zfs%3A1%2Cversion%3Dv2.0&vga=memory%3D16%2Ctype%3Dstd&vmid=100" tf_provider_addr=registry.terraform.io/bpg/proxmox timestamp=2024-04-28T11:55:58.910-0700
2024-04-28T11:55:58.955-0700 [DEBUG] provider.terraform-provider-proxmox_v0.54.0: Received HTTP Response: Expires="Sun, 28 Apr 2024 18:55:58 GMT" Pragma=no-cache tf_http_trans_id=acaf6166-bf97-17be-8871-3e7d8e5b6c4a tf_resource_type=proxmox_virtual_environment_vm tf_http_res_version=HTTP/1.1 tf_mux_provider=tf5to6server.v5tov6Server tf_provider_addr=registry.terraform.io/bpg/proxmox Content-Type=application/json;charset=UTF-8 tf_http_res_status_reason="200 OK" Cache-Control=max-age=0 Content-Length=81 Server=pve-api-daemon/3.0 tf_http_res_status_code=200 tf_req_id=9f849788-b263-09c2-db59-09a8b616d506 tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/logging/logging_http_transport.go:162 @module=proxmox tf_http_op_type=response tf_http_res_body="{\"data\":\"UPID:elodin:0002A693:005854E2:662E9BBE:qmcreate:100:root@pam!opentofu:\"}" Date="Sun, 28 Apr 2024 18:55:58 GMT" timestamp=2024-04-28T11:55:58.954-0700
2024-04-28T11:55:58.955-0700 [DEBUG] provider.terraform-provider-proxmox_v0.54.0: Sending HTTP Request: Accept=application/json Authorization="PVEAPIToken=root@pam!opentofu=da48a39e-2015-4af6-a3d0-94ae09874e74" tf_http_req_body="" tf_http_req_uri="/api2/json/nodes/elodin/tasks/UPID:elodin:0002A693:005854E2:662E9BBE:qmcreate:100:root@pam%21opentofu:/status" tf_http_trans_id=feefa108-8dc9-4c58-5dc5-76994bfd4e52 tf_mux_provider=tf5to6server.v5tov6Server tf_rpc=ApplyResourceChange Host=10.69.13.2:8006 tf_http_req_method=GET tf_provider_addr=registry.terraform.io/bpg/proxmox tf_req_id=9f849788-b263-09c2-db59-09a8b616d506 @module=proxmox User-Agent=Go-http-client/1.1 tf_resource_type=proxmox_virtual_environment_vm @caller=github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/logging/logging_http_transport.go:162 Accept-Encoding=gzip tf_http_op_type=request tf_http_req_version=HTTP/1.1 timestamp=2024-04-28T11:55:58.955-0700
2024-04-28T11:55:58.961-0700 [DEBUG] provider.terraform-provider-proxmox_v0.54.0: Received HTTP Response: tf_http_res_status_code=200 tf_http_res_status_reason="200 OK" tf_provider_addr=registry.terraform.io/bpg/proxmox tf_resource_type=proxmox_virtual_environment_vm Server=pve-api-daemon/3.0 tf_http_res_body="{\"data\":{\"upid\":\"UPID:elodin:0002A693:005854E2:662E9BBE:qmcreate:100:root@pam!opentofu:\",\"pid\":173715,\"node\":\"elodin\",\"status\":\"running\",\"type\":\"qmcreate\",\"pstart\":5788898,\"tokenid\":\"opentofu\",\"user\":\"root@pam\",\"id\":\"100\",\"starttime\":1714330558}}" tf_http_res_version=HTTP/1.1 tf_mux_provider=tf5to6server.v5tov6Server tf_req_id=9f849788-b263-09c2-db59-09a8b616d506 tf_rpc=ApplyResourceChange Connection=[Keep-Alive, Keep-Alive] Content-Length=246 Pragma=no-cache Content-Type=application/json;charset=UTF-8 tf_http_op_type=response Cache-Control=max-age=0 Date="Sun, 28 Apr 2024 18:55:58 GMT" Expires="Sun, 28 Apr 2024 18:55:58 GMT" tf_http_trans_id=feefa108-8dc9-4c58-5dc5-76994bfd4e52 @caller=github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/logging/logging_http_transport.go:162 @module=proxmox timestamp=2024-04-28T11:55:58.961-0700
2024-04-28T11:55:59.962-0700 [DEBUG] provider.terraform-provider-proxmox_v0.54.0: Sending HTTP Request: @caller=github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/logging/logging_http_transport.go:162 User-Agent=Go-http-client/1.1 tf_http_req_uri="/api2/json/nodes/elodin/tasks/UPID:elodin:0002A693:005854E2:662E9BBE:qmcreate:100:root@pam%21opentofu:/status" tf_mux_provider=tf5to6server.v5tov6Server tf_resource_type=proxmox_virtual_environment_vm @module=proxmox Accept=application/json tf_provider_addr=registry.terraform.io/bpg/proxmox tf_req_id=9f849788-b263-09c2-db59-09a8b616d506 tf_rpc=ApplyResourceChange Authorization="PVEAPIToken=root@pam!opentofu=da48a39e-2015-4af6-a3d0-94ae09874e74" Host=10.69.13.2:8006 tf_http_op_type=request Accept-Encoding=gzip tf_http_req_body="" tf_http_req_method=GET tf_http_req_version=HTTP/1.1 tf_http_trans_id=e60ae542-6bfb-e448-5cf7-21fb2e8dd423 timestamp=2024-04-28T11:55:59.961-0700
2024-04-28T11:55:59.966-0700 [DEBUG] provider.terraform-provider-proxmox_v0.54.0: Received HTTP Response: tf_http_res_status_reason="200 OK" Expires="Sun, 28 Apr 2024 18:55:59 GMT" Pragma=no-cache tf_http_op_type=response tf_http_res_status_code=200 @module=proxmox Cache-Control=max-age=0 Server=pve-api-daemon/3.0 tf_req_id=9f849788-b263-09c2-db59-09a8b616d506 tf_resource_type=proxmox_virtual_environment_vm @caller=github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/logging/logging_http_transport.go:162 Date="Sun, 28 Apr 2024 18:55:59 GMT" tf_http_trans_id=e60ae542-6bfb-e448-5cf7-21fb2e8dd423 tf_http_res_version=HTTP/1.1 tf_mux_provider=tf5to6server.v5tov6Server tf_provider_addr=registry.terraform.io/bpg/proxmox tf_rpc=ApplyResourceChange Connection=[Keep-Alive, Keep-Alive] Content-Length=373 Content-Type=application/json;charset=UTF-8 tf_http_res_body="{\"data\":{\"status\":\"stopped\",\"type\":\"qmcreate\",\"starttime\":1714330558,\"user\":\"root@pam\",\"pstart\":5788898,\"upid\":\"UPID:elodin:0002A693:005854E2:662E9BBE:qmcreate:100:root@pam!opentofu:\",\"pid\":173715,\"exitstatus\":\"unable to create VM 100 - unsupported format 'qcow2' at /usr/share/perl5/PVE/Storage/ZFSPoolPlugin.pm line 238.\",\"id\":\"100\",\"tokenid\":\"opentofu\",\"node\":\"elodin\"}}" timestamp=2024-04-28T11:55:59.966-0700
2024-04-28T11:55:59.966-0700 [ERROR] provider.terraform-provider-proxmox_v0.54.0: Response contains error diagnostic: tf_req_id=9f849788-b263-09c2-db59-09a8b616d506 tf_resource_type=proxmox_virtual_environment_vm @module=sdk.proto diagnostic_severity=ERROR tf_proto_version=6.4 tf_provider_addr=registry.terraform.io/bpg/proxmox tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-go@v0.22.2/tfprotov6/internal/diag/diagnostics.go:58 diagnostic_detail="" diagnostic_summary="error waiting for VM creation: task \"UPID:elodin:0002A693:005854E2:662E9BBE:qmcreate:100:root@pam!opentofu:\" failed to complete with exit code: unable to create VM 100 - unsupported format 'qcow2' at /usr/share/perl5/PVE/Storage/ZFSPoolPlugin.pm line 238." timestamp=2024-04-28T11:55:59.966-0700
2024-04-28T11:55:59.985-0700 [DEBUG] State storage *statemgr.Filesystem declined to persist a state snapshot
2024-04-28T11:55:59.985-0700 [ERROR] vertex "proxmox_virtual_environment_vm.nixos_vm" error: error waiting for VM creation: task "UPID:elodin:0002A693:005854E2:662E9BBE:qmcreate:100:root@pam!opentofu:" failed to complete with exit code: unable to create VM 100 - unsupported format 'qcow2' at /usr/share/perl5/PVE/Storage/ZFSPoolPlugin.pm line 238.
╷
│ Error: error waiting for VM creation: task "UPID:elodin:0002A693:005854E2:662E9BBE:qmcreate:100:root@pam!opentofu:" failed to complete with exit code: unable to create VM 100 - unsupported format 'qcow2' at /usr/share/perl5/PVE/Storage/ZFSPoolPlugin.pm line 238.
│ 
│   with proxmox_virtual_environment_vm.nixos_vm,
│   on config.tf.json line 77, in resource.proxmox_virtual_environment_vm.nixos_vm:
│   77:       },
│ 
╵
2024-04-28T11:55:59.995-0700 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2024-04-28T11:55:59.996-0700 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.opentofu.org/bpg/proxmox/0.54.0/linux_amd64/terraform-provider-proxmox_v0.54.0 pid=3851566
```
</details>

--- a/docs/resources/virtual_environment_vm.md
+++ b/docs/resources/virtual_environment_vm.md
@@ -293,7 +293,7 @@ output "ubuntu_vm_public_key" {
     to `ovmf`)
     - `datastore_id` (Optional) The identifier for the datastore to create
         the disk in (defaults to `local-lvm`).
-    - `file_format` (Optional) The file format.
+    - `file_format` (Optional) The file format (defaults to `raw`).
     - `type` (Optional) Size and type of the OVMF EFI disk. `4m` is newer and
         recommended, and required for Secure Boot. For backwards compatibility
         use `2m`. Ignored for VMs with cpu.architecture=`aarch64` (defaults

--- a/fwprovider/tests/resource_vm_test.go
+++ b/fwprovider/tests/resource_vm_test.go
@@ -609,6 +609,30 @@ func TestAccResourceVMDisks(t *testing.T) {
 				RefreshState: true,
 			},
 		}},
+		{"efi disk", []resource.TestStep{
+			{
+				Config: te.renderConfig(`
+				resource "proxmox_virtual_environment_vm" "test_efi_disk" {
+					node_name = "{{.NodeName}}"
+					started   = false
+					name 	  = "test-efi-disk"
+
+					efi_disk {
+						datastore_id = "local-lvm"
+						type = "4m"
+					}
+				}`),
+				Check: resource.ComposeTestCheckFunc(
+					testResourceAttributes("proxmox_virtual_environment_vm.test_efi_disk", map[string]string{
+						"efi_disk.0.datastore_id": "local-lvm",
+						"efi_disk.0.type":         "4m",
+					}),
+				),
+			},
+			{
+				RefreshState: true,
+			},
+		}},
 		{"ide disks", []resource.TestStep{
 			{
 				Config: te.renderConfig(`

--- a/proxmoxtf/resource/vm/vm.go
+++ b/proxmoxtf/resource/vm/vm.go
@@ -72,7 +72,7 @@ const (
 	dvDescription         = ""
 
 	dvEFIDiskDatastoreID                = "local-lvm"
-	dvEFIDiskFileFormat                 = "qcow2"
+	dvEFIDiskFileFormat                 = "raw"
 	dvEFIDiskType                       = "2m"
 	dvEFIDiskPreEnrolledKeys            = false
 	dvTPMStateDatastoreID               = "local-lvm"

